api.c

```diff
@@ -572,10 +571,6 @@ int tcmu_emulate_evpd_inquiry(
 		/* Optimal xfer length */
 		memcpy(&data[12], &val32, 4);
 
-		/* MAXIMUM WRITE SAME LENGTH */
-		val64 = htobe64(VPD_MAX_WRITE_SAME_LENGTH);
-		memcpy(&data[36], &val64, 8);
-
```

Comment:
> Are you saying we should go with your patch where we return 0? If so, for the linux initiator side limitation we would be limited to 0xfff, so are you saying users would need to just adjust that in sysfs or are you saying we should modify the linux initiator side code?

Could also refer this linux commit:
[SCSI] sd: Implement support for WRITE SAME (5db44863b6eb)

We can see one log:
> +       /* Some devices can not handle block counts above 0xffff despite
> +        * supporting WRITE SAME(16). Consequently we default to 64k
> +        * blocks per I/O unless the device explicitly advertises a
> +        * bigger limit.
> +        */

This maybe backwarding compatibility with older or buggy device targets, and the targets could not wholly follow the spec, so the linux code couldn't just set the MAX to zero in vpd.

But I think the users won't have this limitation, so if every thing(such as the initiator) works as the spec defined  and also should be here.

From "SCSI Commands Reference Manual" 
100293068
Rev. D
December 2010

> PAGE LENGTH field
> The PAGE LENGTH field specifies the length of the following VPD page data. The relationship between the PAGE LENGTH field and the CDB ALLOCATION LENGTH field is defined in 2.2.6.
> If the device server supports thin provisioning, then the device server shall set the PAGE LENGTH field to 3Ch.
> If the device server does not support thin provisioning, then the device server may set the PAGE LENGTH field to 10h.

Here it using "may set", not a must or should. So I think the "explicitly" you mentioned should be non-zero and a larger WS MAX LENGTH than default value 0xFFFF.